src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -686,6 +668,21 @@ private static void removeAllUnused(long threshold) throws IOException {
                 }
             }
         }
+
+        public void verifyConnection() throws IOException {
+            synchronized (this) {
+                if (lastVerified > System.currentTimeMillis() - API_URL_REVALIDATE_MILLIS) {
```

Comment:
well here be dragons.   the system clock can go backwards as well as forwards, and may not go forward by as much as you expect despite the amount of time passing.

TL;DR if revalidating is important to you based on some elapsed wall clock time only ever use `System.nanoTime`

Reply:
@jtnord 
Interesting. Found an article giving the trade-offs between the two: https://www.geeksforgeeks.org/java-system-nanotime-vs-system-currenttimemillis/ 

In this case, if verification occurs more or less often than expected in some edge cases that is okay.  We do verification in an attempt to give user meaningful errors when the server they are trying to reach is inaccessible or becomes inaccessible. If it isn't completely accurate that's okay. 

There are other areas where this information makes a difference.  Rate limit calculations for example. 

Thanks, I'll keep this in mind.